File: scratch/compact-guards-integration-20260916/after/src/Bench.go
package Bench

import (
	"fmt"
	"time"
)

var benchmarkEpoch = time.Now()

func BenchNow() float64 {
	return float64(time.Since(benchmarkEpoch).Nanoseconds()) / 1e3
}

func FormatNumber(n float64) string {
	return fmt.Sprintf("%.2f", n)
}

func Opaque(a interface{}) func() interface{} {
	return func() interface{} {
		return opaqueValue(a)
	}
}

// Keep benchmark inputs opaque to the Go compiler without constraining kernels.
//go:noinline
func opaqueValue(a interface{}) interface{} {
	return a
}

File: scratch/record-scalarization-integration-20260916/final/src/Bench.php
<?php

$benchNow = function() {
    return hrtime(true) / 1000.0;
};

$opaque = function($a) {
    return function() use($a) {
        return $a;
    };
};

$formatNumber = function($n) {
    return number_format($n, 2, '.', '');
};

$exports['benchNow'] = $benchNow;
$exports['opaque'] = $opaque;
$exports['formatNumber'] = $formatNumber;

return $exports;
